pkg/frontend/nvme_namespace.go

```diff
@@ -118,29 +127,42 @@ func (s *Server) DeleteNvmeNamespace(_ context.Context, in *pb.DeleteNvmeNamespa
 		return nil, err
 	}
 	// Validate that a resource name conforms to the restrictions outlined in AIP-122.
-	if err := resourcename.Validate(in.Name); err != nil {
+	if err := resourcename.Validate(in.GetName()); err != nil {
 		log.Printf("error: %v", err)
 		return nil, err
 	}
+
+	re := regexp.MustCompile(`nvmeSubsystems/(.+)/nvmeNamespaces/(.+)`)
+	matches := re.FindStringSubmatch(in.GetName())
+	var subsys, ns string
+	if len(matches) >= 3 {
```

Comment:
Almost the same pattern repeats again and again, could we extract that behavior into a dedicated function?